src/dali/transaction_resolver.py

```
@@ -231,62 +231,72 @@ def resolve_transactions(
-
-        if transaction1.unique_id != transaction2.unique_id:
-            raise Exception(f"Internal error: transaction1.{Keyword.UNIQUE_ID.value} != transaction2.{Keyword.UNIQUE_ID.value}: {transaction1}\n{transaction2}")
-        if transaction1.asset != transaction2.asset:
-            raise Exception(f"Internal error: transaction1.{Keyword.ASSET.value} != transaction2.{Keyword.ASSET.value}: {transaction1}\n{transaction2}")
-
-        if isinstance(transaction1, InTransaction) and isinstance(transaction2, OutTransaction):
-            transaction = _resolve_in_out_transaction(transaction1, transaction2, None)
-        elif isinstance(transaction1, OutTransaction) and isinstance(transaction2, InTransaction):
-            transaction = _resolve_out_in_transaction(transaction1, transaction2, None)
-        elif isinstance(transaction1, IntraTransaction) and isinstance(transaction2, IntraTransaction):
-            transaction = _resolve_intra_intra_transaction(transaction1, transaction2, None)
-        else:
-            raise Exception(
-                f"Internal error: attempting to resolve two transactions that aren't Intra/Intra, In/Out or Out/In:\n{transaction1}\n{transaction2}"
-            )
 
-        if read_spot_price_from_web:
-            transaction = _update_spot_price_from_web(transaction, global_configuration)
+            LOGGER.debug("Resolved transaction: %s", str(transaction))
+            resolved_transactions.append(transaction)
 
-        LOGGER.debug("Resolved transaction: %s", str(transaction))
-        resolved_transactions.append(transaction)
+    except KeyboardInterrupt as exc:
+        LOGGER.info("Exiting and saving to cache.")
+        for pair_converter in global_configuration[Keyword.HISTORICAL_PAIR_CONVERTERS.value]:
+            cast(AbstractPairConverterPlugin, pair_converter).save_historical_price_cache()
+        raise Exception("Dali terminated by user.") from exc
```

Raising `Exception` is reserved for internal / sanity errors. In this case I would just rethrow the same exception without modification by using `raise` alone.
``` suggestion
        raise
```